msal/application.py

```diff
@@ -177,7 +177,7 @@ def __init__(
             client_credential=None, authority=None, validate_authority=True,
             token_cache=None,
             http_client=None,
-            verify=True, proxies=None, timeout=None,
+            verify=True, proxies=None, trust_env=False, timeout=None,
```

Comment:
Thanks for your interest in MSAL, @WesleyFransin. However, we hesitate to take this change. The reason being, MSAL is an authentication library, it does not aim to duplicate all the fancy optional parameters that happen to be supported by an underlying http library. So, a generic - albeit less convenient - approach to address these kind of requests is to have the caller provide their own `http_client` with the desirable behavior.